bigtable/client/instance_admin.cc

```diff
@@ -147,5 +154,72 @@ void InstanceAdmin::DeleteCluster(bigtable::InstanceId const& instance_id,
   }
 }
 
+google::bigtable::admin::v2::Cluster InstanceAdmin::CreateClusterImpl(
+    ClusterConfig const& cluster_config,
+    bigtable::InstanceId const& instance_id,
+    bigtable::ClusterId const& cluster_id) {
+  // Copy the policies in effect for the operation.
+  auto rpc_policy = impl_.rpc_retry_policy_->clone();
+  auto backoff_policy = impl_.rpc_backoff_policy_->clone();
+
+  // Build the RPC request, try to minimize copying.
+  auto cluster = cluster_config.as_proto_move();
+  btproto::CreateClusterRequest request;
+  request.set_allocated_cluster(&cluster);
+  request.set_parent(project_name() + "/instances/" + instance_id.get());
+  request.set_cluster_id(cluster_id.get());
+
+  using ClientUtils =
+      bigtable::internal::noex::UnaryClientUtils<InstanceAdminClient>;
+
+  grpc::Status status;
+  auto response = ClientUtils::MakeCall(
+      *impl_.client_, *rpc_policy, *backoff_policy,
+      impl_.metadata_update_policy_, &InstanceAdminClient::CreateCluster,
+      request, "InstanceAdmin::CreateCluster", status, false);
+  if (not status.ok()) {
+    bigtable::internal::RaiseRpcError(status,
+                                      "unrecoverable error in MakeCall()");
+  }
+
+  google::bigtable::admin::v2::Cluster result;
+
+  do {
+    if (response.done()) {
+      if (response.has_response()) {
+        auto const& any = response.response();
+        if (not any.Is<google::bigtable::admin::v2::Cluster>()) {
+          google::cloud::internal::RaiseRuntimeError("invalid result type");
+        }
+        any.UnpackTo(&result);
+        return result;
+      }
+      if (response.has_error()) {
+        bigtable::internal::RaiseRpcError(
+            grpc::Status(static_cast<grpc::StatusCode>(response.error().code()),
+                         response.error().message()),
+            "long running op failed");
+      }
+    }
+    // Wait before polling, and then poll the operation to get the new
+    // "response.
+    auto delay = backoff_policy->on_completion(status);
```

Comment:
Please add a TODO entry saying that we should use the PollingPolicy here once #461 is merged.